gossip/src/crds_value.rs

```diff
@@ -490,84 +495,233 @@ impl Sanitize for NodeInstance {
     }
 }
 
-#[derive(Serialize, Deserialize, Clone, Default, PartialEq, Eq, AbiExample, Debug)]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, AbiExample, AbiEnumVisitor)]
+pub enum SlotsOffsets {
+    RunLengthEncoding(RunLengthEncoding),
+    RawOffsets(RawOffsets),
+}
+
+#[derive(Deserialize, Serialize, Clone, Debug, PartialEq, Eq, AbiExample)]
+pub struct RunLengthEncoding {
+    // For every u8 in encoded, the last bit is the encoded bit, the previous
+    // bits are the count of encoded bit. For example, 0xF1 means there are
+    // (0xF0 >> 1) number of one's (the last bit), so this byte encoded
+    // 120 consecutive 1's. Each byte can encode maximum of 127 consecutive 0's
+    // or 1's.
+    encoded: Vec<u8>,
+    len: usize,
```

Comment:
Why do you need `len: usize` here?
Isn't it true that the decoded RLE has a fixed known length?

Reply:
Removed.